Added Queues message retention period documentation (#19981)
cloudflare/cloudflare-docs: src/content/changelog/queues/2025-02-14-customize-queue-retention-period.mdx

---
title: Customize Queue message retention periods
description: Customize the retention period for a Queue, starting from 60 seconds to a maximum of 14 days.
products:
  - queues
date: 2025-02-14 12:00:00 UTC
---

You can customize a Queue's message retention period, starting from 60s to a maximum of 14 days. Visit the [queues docs](/queues/configuration/configure-queues#queue-configuration) to learn more.

![Customize a Queue's message retention period](~/assets/images/queues/customize-retention-period.png)

By default, Queues retain messages for a default of 4 days. Previously, it was impossible to change the retention period.

With this update, you can retain messages for up to 14 days. Or if you want your consumer to only consume newer messages, you can set the message expiry period to be as short as 60 seconds.

> You can customize the retention period on the Queues settings page on the dashboard, or using Wrangler:

Reviewer: ```suggestion
You can customize the retention period on the Queues settings page on the dashboard or using Wrangler:
```

```bash title="Update message retention period"
$ wrangler queues update my-queue --message-retention-period-secs 600
```

> This feature is available on all new and existing Queues. If you haven't used Cloudflare Queues before, [get started with our guide here](/queues/get-started).

Reviewer: ```suggestion
This feature is available on all new and existing Queues. If you haven't used Cloudflare Queues before, [get started with the Cloudflare Queues guide](/queues/get-started).
```
